src/cmdline/mod.rs

```diff
@@ -516,6 +516,37 @@ impl Cmdline {
     }
 }
 
+/// Convert a Cmdline to String
+///
+/// # Examples
+///
+/// ```rust
+/// # use linux_loader::cmdline::*;
+/// let mut cl = Cmdline::new(20).unwrap();
+/// cl.insert_str("foo").unwrap();
+/// cl.insert_init_args("bar").unwrap();
+/// assert_eq!(String::try_from(&cl).unwrap(), "foo -- bar");
+/// ```
+impl TryFrom<&Cmdline> for String {
+    type Error = Error;
+
+    fn try_from(value: &Cmdline) -> result::Result<Self, Self::Error> {
+        if value.boot_args.is_empty() && value.init_args.is_empty() {
+            Ok("".to_string())
+        } else if value.boot_args.is_empty() {
+            Err(Error::NoBootArgsInserted)
+        } else if value.init_args.is_empty() {
+            Ok(value.boot_args.to_string())
+        } else {
+            Ok(format!(
+                "{}{}{}",
+                value.boot_args, INIT_ARGS_SEPARATOR, value.init_args
+            )
+            .to_string())
```

Comment:
This code ends up being almost the same as the `as_cstring` function. Is there a way to not duplicate the code?

Reply:
Good point. What about this:
``` rust
impl TryFrom<&Cmdline> for CString {
    type Error = Error;

    fn try_from(value: &Cmdline) -> result::Result<Self, Self::Error> {
        let as_string = <String as TryFrom<&Cmdline>>::try_from(value)?;
        Ok(CString::new(as_string).map_err(|_| Error::NullTerminator)?)
    }
}
```